### PR TITLE
Improve linting and formatting

### PR DESCRIPTION
**Pull request by raulrpearson**
In summary, this PR:

- Adds husky pre-commit hook calling lint-staged
- Configures lint-staged. It's installed in the repo root, but currently only runs for changes in the frontend package.
- Cleans up ESLint config and updates versions for eslint and @typescript-eslint. I turned most of the rules into warnings because I'd like to have a discussion and reach an agreement on them, in a different PR.
- Commits reformatted files after running `prettier --write`.

Let me know if you have any questions or if I missed something. Some of the things I'm leaving for future PRs:

- Agreeing on the custom ESLint rules and turning them off or into errors. No point in having a big noisy wall of warnings
- Update dependencies ([Chakra + React 18](https://chakra-ui.com/guides/migration) is finally available, for example)
- Extend linting and [formatting](https://github.com/prettier-solidity/prettier-plugin-solidity) to the hardhat package
- Improve the testing situation, we currently have no tests.

### CODE REVIEW

File: packages/frontend/addresses/1337.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export const ContractAddresses_1337 = {
-  "YieldGate": "0x5fbdb2315678afecb367f032d93f642f64180aa3"
-}
\ No newline at end of file
+  YieldGate: '0x5fbdb2315678afecb367f032d93f642f64180aa3',
```
Comment (sebastianst) [inline packages/frontend/addresses/1337.ts:2]: Would be good to also add a commit fixing the script `packages/hardhat/scripts/deploy.ts` (line 49) that creates these files, because they will always be formatted the wrong way otherwise. But happy to do it in a separate PR.

**Comment by wottpal** [comment]
Hey, I did some changes (with a focus on the hardhat-part):

* For me the husky pre-commit errorred (`npx: command not found`), which is why I adjusted the `.husky/pre-commit` to work properly with `nvm`
* I increased `printWidth` to 100 and set `tabWidth` to 2. Those are settings I personally prefer but we can discuss this.
* I added the same prettier-config to hardhat as well. I first researched how to make a global prettier-file and extend from it in the child-packages but found out there is no way all editors are compatible with. – And as it's only a few lines I thought that's fine ;)
* Improvements regarding the addresses-script I made:
   * Outsourced it in a shared-directory with the aave-addresses
   * Improved code quality
   * Sort imports by chain-id now
   * Run prettier via a `child-process` afterwards to ensure they always look good 💅 

One more thing:
* I would vote for setting all the current custom `eslintrc.json` rules to `off` straight away (they are only 5 and most of them add more distraction than value IMO) – Are you ok with that, @raulrpearson?

File: packages/hardhat/scripts/deploy.ts [hunk 2]
```diff
@@ -1,76 +1,28 @@
...
-  )}}`;
-  const indexFilePath = path.join(addressesDir, "index.ts");
-  fs.writeFileSync(indexFilePath, indexFileContents);
-}
-
-// We recommend this pattern to be able to use async/await everywhere
-// and properly handle errors.
 main()
-  .then(() => process.exit(0))
+  .then(() => {})
```
Comment (sebastianst) [inline packages/hardhat/scripts/deploy.ts:24]: I think you can remove an empty `then`. A promise always runs, even without a `then` afaik.

**Comment by wottpal** [comment]
Hey, I made some further changes and suggested some sane eslint-defaults. Feel free to merge, @raulrpearson if this is working for you, too!

Btw when working with the project in VSCode I suggest opening the workspace-file instead of the directory only. This makes it much easier to switch between sub-packages during work.

* Renamed the workspace-file to `yieldgate.code-workspace` to have a more descriptive window title
* Added the default-formatter as well as format-on-save actions to get everyone's VSCode work properly with our prettier-setup
* Added peacock & workbench colors to separate the window from others (takes only effect if you have the plugin installed anyways)
* Added extension-suggestions (e.g. for `dbaeumer.vscode-eslint` and `esbenp.prettier-vscode`)
* Added some more directories to `.eslintignore` and `.prettierignore`
* Configured husky/lint-staged to also format `.json` files
* Committed a formatted version of all our files with the current settings
* And last but not least this is the eslintrc.json I'm suggesting for now. Feel free to make adjustments and we can change those at any point anyways :)

```json
{ 
  …
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-empty-interface": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "@typescript-eslint/no-non-null-assertion": "warn",
    "react/no-children-prop": "warn",
    "react-hooks/exhaustive-deps": "off",
    "react/jsx-no-target-blank": "off",
    "no-extra-boolean-cast": "off",
    "prefer-const": "warn"
  }
}
```

**Comment by raulrpearson** [comment]
Thanks for the work, @wottpal! I'm not on board with turning off some of those rules, but let's work it out on a different PR to make the review/discussion more manageable.

I added a small formatting commit. Will merge, hope that's okay.
